test/OpenTelemetry.Sampler.AWS.Tests/Utils.cs

```diff
@@ -0,0 +1,60 @@
+// <copyright file="Utils.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Sampler.AWS.Tests;
+
+internal static class Utils
+{
+    internal static SamplingParameters CreateSamplingParametersWithTags(Dictionary<string, string> tags)
+    {
+        ActivityTraceId traceId = ActivityTraceId.CreateRandom();
+        ActivitySpanId parentSpanId = ActivitySpanId.CreateRandom();
+        ActivityTraceFlags traceFlags = ActivityTraceFlags.None;
+
+        var parentContext = new ActivityContext(traceId, parentSpanId, traceFlags);
+
+        var tagList = new List<KeyValuePair<string, object>>();
+
+        foreach (var tag in tags)
+        {
+            tagList.Add(new KeyValuePair<string, object>(tag.Key, tag.Value));
+        }
+
+        return new SamplingParameters(
+            parentContext,
+            traceId,
+            "myActivityName",
+            ActivityKind.Server,
+            tagList,
+            null);
+    }
+
+    internal static Resource CreateResource(string serviceName, string cloudPlatform)
+    {
+        var resourceAttributes = new List<KeyValuePair<string, object>>()
+        {
+            new KeyValuePair<string, object>("service.name", serviceName),
+            new KeyValuePair<string, object>("cloud.platform", cloudPlatform),
+        };
+
+        return ResourceBuilder.CreateEmpty().AddAttributes(resourceAttributes).Build();
+    }
+}
```

Comment:
What do you think about testing the locking mechanism? I think we are fine, but there are concerns about race conditions and dead locks.

Maybe simulate a few hundred/thousand changes here from a mocked backend service?

Reply:
I realized that there were a lot of mutable states in this code which require careful locking. So in my recent code change, I got rid of some of those mutable states and instead create entirely new objects. This should reduce some of the possible race conditions.